### ISSUE TITLE
[Bug]: Installation error after nvimtree refactoring (1f5fd93f80) on rolling branch

### ISSUE THREAD

**Issue by PellePedro** [ChristianChiarulli/LunarVim#1003]
### Problem description

With the factoring of nvimtree in config() and setup() functions I'm experience an error installing due to a null value in default-config.lua

line 162 in `default-config.lua` reads: 
```lua
require("core.nvimtree").config()
```
Before running `PackagerInstall` the statement 
```lua
require("core.nvimtree")
```
 will return null and cause the error when trying to call `config()`
```lua
-- lua/core/nvimtree.lua
local status_ok, nvim_tree_config = pcall(require, "nvim-tree.config")
if not status_ok then
  return
end
```
Note also that the `setup()` function depends requires that` O.plugin.nvimtree` is defined (line 55), and this variable is defined in the `config function()`.

The fix that worked for me was to define a function that performs a call to the plugin, and check the status prior to call the configure().
```lua
-- lua/default-config.lua

local function configure_plugin(plugin_name)
  status_ok, plugin = pcall("require", plugin_name)
  if status_ok then
    plugin.config()
  end
end
configure_plugin("core.nvimtree")
```

I also had to add a check in `lua/core/nvimtree.lua:54` (setup function) as the setup is run at `PackerInstall`.
```lua
-- lua/core/nvimtree.lua:54

if not O.plugin.nvimtree  then
    return
end
```

 

### LunarVim version

rolling ddb627d004d1f6c3bbd593a381ed93bbe54a24fd

### Neovim version (>= 0.5)

v0.6.0-dev+8-g75f758b99

### I have

- [X] Read the readme
- [X] Read the wiki
- [X] Searched for similar issues

**Comment by abzcoding** [comment]
a pull request with those changes would be much appreciated :pleading_face:

**Comment by PellePedro** [comment]
My first pull request to LunarVim :) https://github.com/ChristianChiarulli/LunarVim/pull/1004

**Comment by abzcoding** [comment]
Fixed by #1012
